src/charm.py

```diff
@@ -80,10 +82,52 @@ def _on_config_changed(self, _):
         else:
             self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
 
+    def _create_service_account(self):
+        """Create a ServiceAccount for training-operator."""
+        self.unit.status = MaintenanceStatus("Creating ServiceAccount")
+        try:
+            K8sResources.create_service_account()
+        except ResourceHandlingFailed as e:
+            logging.error(
+                "Unable to create ServiceAccount for training-operator: %s", str(e)
+            )
+        else:
+            logging.info("Successfully created ServiceAccount for training-operator")
+            self.unit.status = ActiveStatus()
+
+    def _create_cluster_role(self):
+        """Create a ClusterRole for training-operator."""
+        self.unit.status = MaintenanceStatus("Creating ClusterRole")
+        try:
+            K8sResources.create_cluster_role()
+        except ResourceHandlingFailed as e:
+            logging.error(
+                "Unable to create ClusterRole for training-operator: %s", str(e)
+            )
+        else:
+            logging.info("Successfully created ClusterRole for training-operator")
+            self.unit.status = ActiveStatus()
+
+    def _create_cluster_role_binding(self):
+        """Create a ClusterRoleBinding for training-operator."""
+        self._create_cluster_role()
+        self.unit.status = MaintenanceStatus("Creating ClusterRole")
+        try:
+            K8sResources.create_cluster_role_binding()
+        except ResourceHandlingFailed as e:
+            logging.error(
+                "Unable to create ClusterRoleBinding for training-operator: %s", str(e)
+            )
+        else:
+            logging.info(
+                "Successfully created ClusterRoleBinding for training-operator"
+            )
+            self.unit.status = ActiveStatus()
 
     def _on_install(self, _):
-        """Placeholder for install event"""
-        pass
+        """Event handler for on-install events"""
+        self._create_service_account()
+        self._create_cluster_role_binding()
```

Comment:
If you do indeed make these helpers return something, you could do something like this:

```suggestion
        sa = self._create_service_account()
        crb = self._create_cluster_role_binding()
        
        if sa and crb:
            self.unit.status = ActiveStatus()
        else:
            self.unit.status = BlockedStatus("unable to create necessary resources, check logs")
```